packages/target-graph/tests/transitiveReduction.test.ts

```diff
@@ -0,0 +1,114 @@
+import { TargetGraphBuilder } from "../src/TargetGraphBuilder";
+import { Target } from "../src/types/Target";
+import { transitiveReduction } from "../src/transitiveReduction";
+import { diff } from "jest-diff";
+
+describe("transitiveReduction", () => {
+  it("should reduce the graph transitively", () => {
+    const target1 = createTarget("a", "build", 1);
+    const target2 = createTarget("b", "build", 1);
+    const target3 = createTarget("c", "build", 1);
+    const target4 = createTarget("d", "build", 1);
+
+    const builder = new TargetGraphBuilder();
+    builder.addTarget(target1);
+    builder.addTarget(target2);
+    builder.addTarget(target3);
+    builder.addTarget(target4);
+    builder.addDependency(target1.id, target2.id);
+    builder.addDependency(target1.id, target3.id);
+    builder.addDependency(target2.id, target3.id);
+    builder.addDependency(target4.id, target1.id);
+
+    const targetGraph = builder.subgraph(["c#build"]);
+
+    const targets = [...targetGraph.targets.values()];
+
+    expect(
+      diff(simplify(targets), simplify(transitiveReduction(targets)), { aAnnotation: "Original", bAnnotation: "Transitively Reduced" })
+    ).toMatchInlineSnapshot(`
+      "- Original
+      + Transitively Reduced
+
+        Array [
+          Object {
+            "dependencies": Array [],
+            "dependents": Array [
+              "c#build",
+              "a#build",
+              "b#build",
+              "d#build",
```

Comment:
it just occurred to me that we never fixed up the dependents array after removing these nodes!